### PR TITLE
Strip leading slash to allow testing repository with class keyword

### PR DESCRIPTION
**Pull request by rgeraads**
When testing a repository that has the `::class` keyword instead of a string representation, a test failed because that expected a leading slash.

The fix allows for both FQN with and without leading slash so it is BC

Thanks @reenl for helping me out!

### CODE REVIEW

**Comment by rgeraads** [comment]
```
0.15s$ phpunit --exclude-group=none
PHPUnit 6.3.0 by Sebastian Bergmann and contributors.
Time: 87 ms, Memory: 10.00MB
No tests executed!
```
😟

**Comment by reenl** [comment]
Thanks for the mention! Seems master is still running phpunit 4.8, however that one crashes because it is not able to mock return types.

**Comment by othillo** [comment]
check this PR: https://github.com/broadway/broadway/pull/331

**Comment by rgeraads** [comment]
I can rebase my changes to that PR if you want

File: test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php [hunk 1]
```diff
@@ -146,7 +146,7 @@ public function it_calls_the_event_stream_decorators_with_the_correct_arguments(
         $lastCall = $this->eventStreamDecorator->getLastCall();
 
         $this->assertEquals($aggregate->getAggregateRootId(), $lastCall['aggregateIdentifier']);
-        $this->assertEquals('\\' . get_class($aggregate), $lastCall['aggregateType']);
+        $this->assertEquals(get_class($aggregate), $this->stripLeadingSlash($lastCall['aggregateType']));
```
Comment (othillo) [inline test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php]: so you're saying https://github.com/broadway/broadway/blob/master/src/Broadway/EventSourcing/EventSourcingRepository.php#L51 allows classnames with slashes and without?
This is validated by assertExtendsEventSourcedAggregateRoot  -> Assert::subclassOf which in turn calls PHP's is_subclass_of function in https://github.com/beberlei/assert/blob/master/lib/Assert/Assertion.php#L1306

wouldn't it be easier to just use assertTrue(is_subclass_of($aggregate, $lastCall['aggregateType'])) ?
or maybe even assertInstanceOf($lastCall['aggregateType'], $aggregate) ?

Comment (othillo) [inline test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php]: Would you mind adding a test for both situations for sake of regression testing?

Comment (rgeraads) [reply test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php]: I don't understand how, since this already is a test

Comment (othillo) [reply test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php]: You mentioned it breaks when creating a repository using aggregate::class notation and not when using the string FQDN representation. I would like to see a test for both conditions. 

But I've tried it manually by modifying https://github.com/broadway/broadway/blob/master/test/Broadway/EventSourcing/EventSourcingRepositoryTest.php#L29 and your change works, so I'm ok with it.

**Comment by othillo** [comment]
thank you, @rgeraads !

**Pull request merged**